lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -1111,6 +1111,7 @@ def get_location_info(self, location: str, log: Logger) -> AzureLocation:
                     except Exception as e:
                         log.error(f"unknown sku: {sku_obj}")
                         raise e
+            plugin_manager.hook.azure_update_vm_capabilities(capabilities=all_skus)
```

Comment:
How about put it in the condition `should_refresh`? it will improve performance a lot.